usr/src/uts/i86pc/io/vmm/vmm.c

```diff
@@ -3721,3 +3732,17 @@ vm_ioport_unhook(struct vm *vm, void **cookie)
 
 	*cookie = NULL;
 }
+
+int
+vm_arc_resv(struct vm *vm, uint64_t len)
+{
+	int err = 0;
+
+	err = arc_virt_machine_reserve((size_t)(len >> PAGE_SHIFT));
```

Comment:
len >> PAGE_SHIFT or something more sophisticated?  (Simple corner-case... what if I ask for PAGESIZE + 512?)